batteries/argparse.luau

```diff
@@ -0,0 +1,137 @@
+local argparse = {}
+argparse.__index = argparse
+
+type ArgKind = "POSITIONAL" | "FLAG" | "OPTION"
+type ArgOptions = {
+	help: string?,
+	aliases: { string }?,
+	default: string?,
+	required: boolean?,
+}
+
+type ArgData = {
+	name: string,
+	kind: ArgKind,
+	options: ArgOptions,
+}
+
+type ParseResult = {
+	values: { [string]: string },
+	flags: { [string]: boolean },
+	fwdArgs: { string },
+}
+
+function argparse.new()
+	local self = setmetatable({}, argparse)
+	self.arguments = {}
+	self.positional = {}
+	self.parsed = { values = {}, flags = {}, fwdArgs = {} }
+	return self
+end
+
+function argparse:add(name, kind, options)
+	local argument = {
+		name = name,
+		kind = kind,
+		options = options or { aliases = {}, required = false },
+	}
+	table.insert(self.arguments, argument)
+
+	if kind == "POSITIONAL" then
+		table.insert(self.positional, argument)
+	end
+end
+
+function argparse:parse(args)
+	local i = 1
+	local pos_index = 1
+	while i <= #args do
+		local arg = args[i]
+		if arg == "--" then
+			for j = i + 1, #args do
+				table.insert(self.parsed.fwdArgs, args[j])
+			end
+			break
+		elseif string.sub(arg, 1, 2) == "--" then
```

Comment:
```suggestion
		if arg == "--" then
			for j = i + 1, #args do
				table.insert(self.parsed.fwdArgs, args[j])
			end
			break
		end
		
		if string.sub(arg, 1, 2) == "--" then
```

Reply:
Above is an example, but generally, I'd like to rewrite this to use early-break and/or early-return rather than just a big if-elseif chain with nested conditionals under it.